app/src/main/java/com/infomaniak/drive/ui/fileList/fileShare/FileShareDetailsFragment.kt

```diff
@@ -182,59 +192,67 @@ class FileShareDetailsFragment : Fragment() {
             file = file,
             shareLink = shareLink,
-            onTitleClicked = { newShareLink, currentFileId ->
-                this.shareLink = newShareLink
-                val (permissionsGroup, currentPermission) = FileDetailsInfoFragment.selectPermissions(
-                    isFolder = file.isFolder(),
-                    isOnlyOffice = file.onlyoffice,
-                    shareLinkExist = this.shareLink != null,
-                )
-                safeNavigate(
-                    FileShareDetailsFragmentDirections.actionFileShareDetailsFragmentToSelectPermissionBottomSheetDialog(
-                        currentFileId = currentFileId,
-                        currentPermission = currentPermission,
-                        permissionsGroup = permissionsGroup,
-                    )
-                )
-            },
-            onSettingsClicked = { newShareLink, currentFile ->
-                this.shareLink = newShareLink
-                safeNavigate(
-                    FileShareDetailsFragmentDirections.actionFileShareDetailsFragmentToFileShareLinkSettings(
-                        fileId = currentFile.id,
-                        driveId = currentFile.driveId,
-                        shareLink = newShareLink,
-                        isOnlyOfficeFile = currentFile.onlyoffice,
-                        isFolder = currentFile.isFolder(),
-                    )
-                )
-            })
+            onTitleClicked = { newShareLink -> handleOnShareLinkTitleClicked(newShareLink) },
+            onSettingsClicked = { newShareLink -> handleOnShareLinkSettingsClicked(newShareLink) })
+    }
+
+    private fun handleOnShareLinkTitleClicked(newShareLink: ShareLink?) {
+        shareLink = newShareLink
+        val (permissionsGroup, currentPermission) = FileDetailsInfoFragment.selectPermissions(
+            isFolder = file.isFolder(),
+            isOnlyOffice = file.onlyoffice,
+            shareLinkExist = shareLink != null,
+        )
+        safeNavigate(
+            FileShareDetailsFragmentDirections.actionFileShareDetailsFragmentToSelectPermissionBottomSheetDialog(
+                currentFileId = file.id,
+                currentPermission = currentPermission,
+                permissionsGroup = permissionsGroup,
+            )
+        )
+    }
+
+    private fun handleOnShareLinkSettingsClicked(newShareLink: ShareLink) {
+        shareLink = newShareLink
+        safeNavigate(
+            FileShareDetailsFragmentDirections.actionFileShareDetailsFragmentToFileShareLinkSettings(
+                fileId = file.id,
+                driveId = file.driveId,
+                shareLink = newShareLink,
+                isOnlyOfficeFile = file.onlyoffice,
+                isFolder = file.isFolder(),
+            )
+        )
+    }
+
+    private fun showShareLinkView() {
+        shareLinkLayout.isVisible = true
     }
 
     private fun hideShareLinkView() {
         shareLinkLayout.isGone = true
     }
```

Comment:
Duplicate here, just one method is sufficient here.
For example: 
```suggestion
private fun showShareLinkView(show: Boolean) {
        shareLinkLayout.isVisible = show
    }
```